core/src/main/java/org/mskcc/cbio/portal/util/SparkConfiguration.java

```diff
@@ -0,0 +1,50 @@
+package org.mskcc.cbio.portal.util;
+
+import org.mskcc.cbio.portal.util.GlobalProperties;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.SparkSession;
+
+@Configuration
+public class SparkConfiguration {
+
+    private String appName = GlobalProperties.getProperty("spark.app.name");
```

Comment:
better to use spring annotation for properties. example
```
@Value("${spark.app.name}")
private String appName;
```

Reply:
Agreed. thanks for the comment.

Reply:
I tried changing it to @Value and can't get it to work after trying (@Component, @Configuration etc). I see that there is no @Value used in org.mskcc.cbio.portal package outside of GlobalProperties and properties are referred from GlobalProperties. Is there a reason behind this? I'm using @Value in other java classes (not in this PR) in persistence layer with no problem.

Reply:
weird. may be for now you could leave as it is

Reply:
You don't see @value in most of the classes because most packages in the core module are not under spring control.  If spring doesn't create an instance of the class, you cannot auto-wire or inject other values into it.  However, I do see in applicationContext-core.xml that the base package org.mskcc.cbio.portal is scanned for components so you should be able to get values injected into it.  Whats missing in applicationContext-core.xml is a propertyPlaceholderConfigurer which will provides injection of properties from portal.properties.  You want to add the following bean into applicationContext-core.xml:

[applicationContext-web.xml](https://github.com/cBioPortal/cbioportal/blob/master/web/src/main/resources/applicationContext-web.xml#L14)

If that doesn't work, it may be because you are missing a setter method for the property.